libs/ledger-live-common/src/families/internet_computer/bridge/utils/icp/index.ts

```diff
@@ -0,0 +1,156 @@
+import Transport from "@ledgerhq/hw-transport";
+import { Account } from "@ledgerhq/types-live";
+import { Transaction } from "../../../types";
+import ICP from "@zondax/ledger-icp";
+import { constructionInvoke, getICPRosettaNetworkIdentifier } from "../network";
+import {
+  ConstructionCombineRequest,
+  ConstructionCombineResponse,
+  ConstructionHashRequest,
+  ConstructionHashResponse,
+  ConstructionPayloadsRequest,
+  ConstructionPayloadsResponse,
+  ConstructionSubmitRequest,
+  ConstructionSubmitResponse,
+} from "../types";
+import {
+  ingressExpiry,
+  generateOperations,
+  generateSignaturesPayload,
+} from "./utils";
+import { Cbor } from "@dfinity/agent";
+import { Principal } from "@dfinity/principal";
+import { isError } from "../../../utils";
+import BigNumber from "bignumber.js";
+
+if (typeof BigInt === "undefined") {
+  global.BigInt = require("big-integer");
+}
+
+export const getUnsignedTransaction = async (
+  transaction: Transaction,
+  account: Account
+): Promise<{
+  unsignedTxn: string;
+  payloads: ConstructionPayloadsResponse["payloads"];
+}> => {
+  const ops = generateOperations(transaction, account);
+  const pubkeys = [
+    {
+      hex_bytes: account.xpub ?? "",
+      curve_type: "secp256k1",
+    },
+  ];
+
+  const reqOpts: ConstructionPayloadsRequest = {
+    ...getICPRosettaNetworkIdentifier(),
+    operations: ops,
+    public_keys: pubkeys,
+    metadata: {
+      memo: parseInt(transaction.memo ?? "0"),
+    },
+  };
+  const { payloads, unsigned_transaction } = await constructionInvoke<
+    ConstructionPayloadsRequest,
+    ConstructionPayloadsResponse
+  >(reqOpts, "payloads");
+
+  return { unsignedTxn: unsigned_transaction, payloads };
+};
+
+export const signICPTransaction = async ({
+  unsignedTxn,
+  transport,
+  path,
+  payloads,
+  pubkey,
+}: {
+  unsignedTxn: string;
+  transport: Transport;
+  path: string;
+  payloads: ConstructionPayloadsResponse["payloads"];
+  pubkey: string;
+}): Promise<{
+  signatures: { txnSig: string; readSig: string };
+  signedTxn: string;
+}> => {
+  const icp = new ICP(transport as any);
+  const decodedTxn: any = Cbor.decode(Buffer.from(unsignedTxn, "hex"));
```

Comment:
`any` is discouraged as a general rule. Any chance this can by typed? Would be better, but not necessary at the moment.

Reply:
I see, I completely agree... Fixed. It's ruins of early development haha 😅